core/src/bls_sigverify/stats.rs

```diff
@@ -0,0 +1,311 @@
+use std::{
+    sync::atomic::{AtomicU64, Ordering},
+    time::{Duration, Instant},
+};
+
+pub(crate) const STATS_INTERVAL_DURATION: Duration = Duration::from_secs(1);
+
+#[derive(Debug, Default)]
+pub(crate) struct BLSPacketStats {
+    pub(crate) recv_batches_us_hist: histogram::Histogram, // time to call recv_batch
+    pub(crate) verify_batches_pp_us_hist: histogram::Histogram, // per-packet time to call verify_batch
+    pub(crate) dedup_packets_pp_us_hist: histogram::Histogram, // per-packet time to call verify_batch
+    pub(crate) batches_hist: histogram::Histogram, // number of packet batches per verify call
+    pub(crate) packets_hist: histogram::Histogram, // number of packets per verify call
+    pub(crate) num_deduper_saturations: usize,
+    pub(crate) total_batches: usize,
+    pub(crate) total_packets: usize,
+    pub(crate) total_dedup: usize,
+    pub(crate) total_dedup_time_us: usize,
+    pub(crate) total_verify_time_us: usize,
+}
+
+impl BLSPacketStats {
+    pub(crate) fn maybe_report(&self) {
+        // No need to report a datapoint if no batches/packets received
+        if self.total_batches == 0 {
+            return;
+        }
+
+        datapoint_info!(
+            "tpu-consensus-messages",
```

Comment:
seems like we should have `verify` or `verifier` or something in the name here for discoverability